src/isar_turtlebot/robotinterface.py

```diff
@@ -27,36 +34,45 @@ def __init__(self):
 
     def schedule_step(self, step: Step) -> Tuple[bool, Optional[Any], Optional[Joints]]:
 
+        previous_run_id: str = self.get_run_id()
+        self.publish_task(step=step)
+        run_id: str = self.wait_for_updated_task(previous_run_id=previous_run_id)
+        return True, run_id, None
+
+    def publish_task(self, step: Step) -> None:
         if isinstance(step, DriveToPose):
-            previous_run_id: str = self.get_run_id()
-            pose_message: dict = {
-                "goal": {
-                    "target_pose": {
-                        "header": {
-                            "seq": 0,
-                            "stamp": {"secs": 1533, "nsecs": 746000000},
-                            "frame_id": "map",
+            self.publish_navigation_task(pose=step.pose)
+
+        elif isinstance(step, (TakeImage, TakeThermalImage)):
+            raise TypeError(f"Image step is not  yet implemented on turtlebot")
+
+    def publish_navigation_task(self, pose: Pose) -> None:
```

Comment:
Is this beyond the scope of the commit? If so, consider splitting the commit or rewording it to incorporating the schedule step as well as subscribing to the pose topic.

Reply:
Changed the commit message accordingly